agent/src/main/java/com/newrelic/agent/android/NewRelic.java

```diff
@@ -472,9 +472,11 @@ public static void recordMetric(String name, String category, int count, double
         StatsEngine.notice().inc(MetricNames.SUPPORTABILITY_API
                 .replace(MetricNames.TAG_NAME, "recordMetric"));
 
-        log.debug("NewRelic.recordMetric invoked for name " + name + ", category: " + category +
-                ", count: " + count + ", totalValue " + totalValue + ", exclusiveValue: " + exclusiveValue +
-                ", countUnit: " + countUnit + ", valueUnit: " + valueUnit);
+        StringBuilder logString = new StringBuilder();
```

Comment:
How does this reduce excessive string concatenation? IIRC, the complaint was the agent doing the work to build this excessive `DEBUG` level log message, even when logLevel was set to something below DEBUG. The change, imo, is to wrap the call in a conditional that checks the current log level first, before constructing the message.

Reply:
String builder is more efficient when two or more strings are concat, i agree need to add log level before concatenation happen.